documentation/modules/vmware-prerequisites.adoc

```diff
@@ -14,10 +14,16 @@ The following prerequisites apply to VMware migrations:
 * You must create a VMware Virtual Disk Development Kit (VDDK) image.
 * You must obtain the SHA-1 fingerprint of the vCenter host.
 * If you are migrating more than 10 VMs from an ESXi host in the same migration plan, you must increase the NFC service memory of the host.
+* It is strongly recommended to disable hibernation because {project-first} does not support migrating hibernated VMs.
+
+[IMPORTANT]
+====
+If there is a power outage, data might be lost for a VM for which hibernation is disabled. However, if hibernation is not disabled, migration will fail.
```

Comment:
@RichardHoch - i think this is very good; clear and concise with no ambiguity

@ahadas - you good with this?

Reply:
I'm still not sure about the last part - as far as I know we don't require to disable it.
@qiyuann could you please keep me honest - that migration succeeds for a VM for which hibernation is enabled but the VM is not in hibernated state?

Reply:
I spoke with Nisim yesterday and apparently I didn't understand it correctly - that "hibernation" setting is not about whether the VM can be hibernated but what's the operation that is executed when pressing the power-off button and as such it's not possible to shutdown a vm "normally" when hibernation is enabled. in that case, this phrasing makes sense. @nyoxi are you aware of this?

Reply:
One cannot expect that hibernated VM will come up properly after being migrated. Hibernated OS does not tolerate too many hardware changes (Would you hibernate your OS before replacing mainboard in your bare metal machine?) and changes to the disk content are not necessarily safe either. Not to mention that you are likely to lose your saved RAM depending on the configuration of your OS. This also includes the fast boot/fast start feature of Windows where virt-v2v is not able to modify the filesystem and registry. What we should be telling people is to make sure their guest OS is properly shut down.

To extend on that, in IMS we had a document with "best practices" (Do we have something like that too?) where we were suggesting to not only shut down the guest but to perform a reboot before doing the shut down for migration. This is to make sure the OS is in a proper condition and to avoid any surprises after migration which could be mistaken as bugs in our tooling. There be partially installed apps waiting for restart or system updates in queue or just misconfigured services.

With that all being said, I don't thing we convey the necessary message to the users/admins.

Reply:
> Not to mention that you are likely to lose your saved RAM depending on the configuration of your OS.

On a second thought, this should not really happen.